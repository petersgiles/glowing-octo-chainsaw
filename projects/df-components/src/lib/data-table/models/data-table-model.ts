export class Page {
  // The number of elements in the page
  public size: number = 0
  // The total number of elements
  public totalElements: number = 0
  // The total number of pages
  public totalPages: number = 0
  // The current page number
  public pageNumber: number = 0
}

// tslint:disable-next-line:max-classes-per-file
export class PagedData<T> {
  public data = []
  public page = new Page()
}
