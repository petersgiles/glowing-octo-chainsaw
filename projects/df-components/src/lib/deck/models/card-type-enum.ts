export enum CardType {
    Standard = "STANDARD",
    Parent = "PARENT",
    Wiki = "WIKI",
    Markdown = "MARKDOWN",
    Audio = "AUDIO",
    Video = "VIDEO",
    Embed = "EMBED",
    Frame = "FRAME",
    Chart = "CHART",
    Image = "IMAGE",
    Newtab = "NEWTAB",
    Refiner = "REFINER",
    BriefSummary = "BRIEFSUMMARY"
  }