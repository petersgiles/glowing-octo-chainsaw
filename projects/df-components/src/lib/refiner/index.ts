export * from "./refiner.module"
export * from "./refiner.models"
