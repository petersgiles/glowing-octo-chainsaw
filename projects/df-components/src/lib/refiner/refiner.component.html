<mdc-list dense *ngFor="let group of refinerGroups" class="list">
  <mdc-list-item
    class="digital-first-list-heading"
    (click)="group.expanded = !group.expanded"
  >
    {{ group.title }}
    <mdc-icon mdcListItemMeta
      >keyboard_arrow_{{ group.expanded ? "down" : "right" }}</mdc-icon
    >
  </mdc-list-item>
  <div [hidden]="!group.expanded || !group.refiners">
    <mdc-list-item
      class="refiner"
      *ngFor="let refiner of group.refiners"
      (click)="handleClickRefiner(refiner)"
    >
      <mdc-icon
        >check_box{{ refiner.selected ? "" : "_outline_blank" }}</mdc-icon
      >
      {{ refiner.title }}
      <span mdcListItemMeta class="refiner-metadata">{{ refiner.count }}</span>
    </mdc-list-item>
  </div>
</mdc-list>
