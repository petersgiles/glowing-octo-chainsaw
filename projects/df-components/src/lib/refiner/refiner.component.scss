@import "../../theme";

.list {
  cursor: pointer;
}

.refiner {
  padding: 0 0 0 0;
  text-overflow: ellipsis;
}

.refiner-metadata {
  padding: 11px;
}

.digital-first-list-heading {
  @include heading-color();
}