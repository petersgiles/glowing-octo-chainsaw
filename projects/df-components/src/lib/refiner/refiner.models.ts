export interface Refiner {
  id: string
  title: string
  selected?: boolean
  count?: number
}

export interface RefinerGroup {
  id: string
  title: string
  refiners: Refiner[]
  expanded?: boolean
  count?: number
}
