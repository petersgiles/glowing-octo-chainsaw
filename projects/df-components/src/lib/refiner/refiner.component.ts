import { Component, EventEmitter, Input, Output } from "@angular/core"
import { Refiner, RefinerGroup } from "./refiner.models"

@Component({
  selector: "df-refiner",
  templateUrl: "./refiner.component.html",
  styleUrls: ["./refiner.component.scss"]
})
export class RefinerComponent {
  @Input()
  public refinerGroups: RefinerGroup[]

  @Output()
  public refine: EventEmitter<RefinerGroup[]> = new EventEmitter()

  public handleClickRefiner(refiner: Refiner) {
    refiner.selected = !refiner.selected
    this.refine.emit(
      this.refinerGroups
        .map(group => ({
          ...group,
          refiners: group.refiners.filter(r => r.selected)
        }))
        .filter(group => group.refiners.length)
    )
  }
}
