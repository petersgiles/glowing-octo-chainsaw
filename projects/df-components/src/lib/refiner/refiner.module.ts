import { CommonModule } from '@angular/common'
import { NgModule } from '@angular/core'

import { MdcIconModule, MdcListModule, MdcTextFieldModule } from '@angular-mdc/web'
import { RefinerComponent } from './refiner.component'

const COMPONENTS = [
  RefinerComponent,
]

@NgModule({
  imports: [
    CommonModule,
    MdcListModule,
    MdcIconModule,
    MdcTextFieldModule,
  ],
  declarations: COMPONENTS,
  exports: COMPONENTS,
})
export class RefinerModule { }
