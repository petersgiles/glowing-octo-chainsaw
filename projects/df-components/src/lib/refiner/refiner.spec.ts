import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RefinerComponent } from './refiner.component';
import { DebugElement, NO_ERRORS_SCHEMA } from '@angular/core';
import { RefinerGroup } from './refiner.models';
import { By } from '@angular/platform-browser';

describe('RefinerComponent', () => {
    let component: RefinerComponent;
    let fixture: ComponentFixture<RefinerComponent>;
    let firstRefiner: DebugElement

    beforeEach(async(() => {
        TestBed.configureTestingModule({
            schemas: [NO_ERRORS_SCHEMA],
            declarations: [RefinerComponent]
        })
            .compileComponents();
    }));

    beforeEach(() => {
        fixture = TestBed.createComponent(RefinerComponent);
        component = fixture.componentInstance;
        component.refinerGroups = testRefinerGroups
        fixture.detectChanges();

        firstRefiner = fixture.debugElement.query(By.css('.refiner'))
    });

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('should display lists of refiners', () => {
        expect(component).toMatchSnapshot()
    });

    it('should emit all selected refiners', () => {
        spyOn(component.refine, 'emit')
        firstRefiner.nativeElement.click()
        expect(component.refine.emit).toHaveBeenCalledWith([
            {
                ...testRefinerGroups[0],
                refiners: [testRefinerGroups[0].refiners[0]]
            }
        ])
    });
});


const testRefinerGroups: RefinerGroup[] = [
    {
        "id": "cb80b0e8-b441-49da-9c46-41e5b3a32102",
        "title": "Genera",
        "refiners": [
            {
                "id": "5f1e5511-7c4b-4df8-8711-b646133cca47",
                "title": "Actinidia",
                "count": 1
            },
            {
                "id": "1d639213-51bd-4f56-b370-fa2656c00e94",
                "title": "Allium",
                "count": 2
            },
            {
                "id": "d3f93714-766f-44bd-8233-eade778feb7c",
                "title": "Anacardium",
                "count": 1
            }
        ]
    }
    ,
    {
        "id": "5c6813d3-83fa-43cd-9023-996f9aac8931",
        "title": "Starts with",
        "refiners": [
            {
                "id": "7db2031a-fa8c-4cd2-badd-c48b299a58c3",
                "title": "A",
                "count": 3
            },
            {
                "id": "5730be84-c3b9-4caf-affa-fee468b85778",
                "title": "B",
                "count": 5
            },
            {
                "id": "7f65744c-0721-40ed-8e02-940ed303ec46",
                "title": "C",
                "count": 10
            }
        ]
    }
]