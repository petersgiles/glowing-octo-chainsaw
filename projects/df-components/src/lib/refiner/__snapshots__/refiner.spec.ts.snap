// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefinerComponent should display lists of refiners 1`] = `
RefinerComponent {
  "refine": EventEmitter {
    "__isAsync": false,
    "_isScalar": false,
    "closed": false,
    "hasError": false,
    "isStopped": false,
    "observers": Array [],
    "thrownError": null,
  },
  "refinerGroups": Array [
    Object {
      "id": "cb80b0e8-b441-49da-9c46-41e5b3a32102",
      "refiners": Array [
        Object {
          "count": 1,
          "id": "5f1e5511-7c4b-4df8-8711-b646133cca47",
          "title": "Actinidia",
        },
        Object {
          "count": 2,
          "id": "1d639213-51bd-4f56-b370-fa2656c00e94",
          "title": "Allium",
        },
        Object {
          "count": 1,
          "id": "d3f93714-766f-44bd-8233-eade778feb7c",
          "title": "Anacardium",
        },
      ],
      "title": "Genera",
    },
    Object {
      "id": "5c6813d3-83fa-43cd-9023-996f9aac8931",
      "refiners": Array [
        Object {
          "count": 3,
          "id": "7db2031a-fa8c-4cd2-badd-c48b299a58c3",
          "title": "A",
        },
        Object {
          "count": 5,
          "id": "5730be84-c3b9-4caf-affa-fee468b85778",
          "title": "B",
        },
        Object {
          "count": 10,
          "id": "7f65744c-0721-40ed-8e02-940ed303ec46",
          "title": "C",
        },
      ],
      "title": "Starts with",
    },
  ],
}
`;
